test/es-module/test-esm-cjs-load-error-note.mjs

```diff
@@ -1,163 +1,96 @@
-pError1.stderr.on('data', (data) => {
-  pError1Stderr += data;
-});
-pError1.on('close', mustCall((code) => {
-  assert.strictEqual(code, expectedCode);
-  assert.ok(pError1Stderr.includes('Error: some error'));
-  assert.ok(!pError1Stderr.includes(expectedNote),
-            `${expectedNote} must not be included in ${pError1Stderr}`);
-}));
-
-const pError2 = spawn(process.execPath, [Error2]);
-let pError2Stderr = '';
-pError2.stderr.setEncoding('utf8');
-pError2.stderr.on('data', (data) => {
-  pError2Stderr += data;
-});
-pError2.on('close', mustCall((code) => {
-  assert.strictEqual(code, expectedCode);
-  assert.ok(pError2Stderr.includes('string'));
-  assert.ok(!pError2Stderr.includes(expectedNote),
-            `${expectedNote} must not be included in ${pError2Stderr}`);
-}));
-
-const pError3 = spawn(process.execPath, [Error3]);
-let pError3Stderr = '';
-pError3.stderr.setEncoding('utf8');
-pError3.stderr.on('data', (data) => {
-  pError3Stderr += data;
-});
-pError3.on('close', mustCall((code) => {
-  assert.strictEqual(code, expectedCode);
-  assert.ok(pError3Stderr.includes('null'));
-  assert.ok(!pError3Stderr.includes(expectedNote),
-            `${expectedNote} must not be included in ${pError3Stderr}`);
-}));
-
-const pError4 = spawn(process.execPath, [Error4]);
-let pError4Stderr = '';
-pError4.stderr.setEncoding('utf8');
-pError4.stderr.on('data', (data) => {
-  pError4Stderr += data;
+const mustIncludeMessage = {
+  getMessage: () => (stderr) => `${expectedNote} not found in ${stderr}`,
+  includeNote: true,
+};
+const mustNotIncludeMessage = {
+  getMessage: () => (stderr) => `${expectedNote} must not be included in ${stderr}`,
+  includeNote: false,
+};
+
+describe('ESM: Errors for unexpected exports', { concurrency: true }, () => {
+  for (
+    const { errorNeedle, filePath, getMessage, includeNote }
+    of [
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-unexpected-export-1.cjs'),
+        ...mustIncludeMessage,
+      },
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-unexpected-import-1.cjs'),
+        ...mustIncludeMessage,
+      },
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-promiserej-import-2.cjs'),
+        ...mustNotIncludeMessage,
+      },
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-unexpected-import-3.cjs'),
+        ...mustIncludeMessage,
+      },
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-unexpected-import-4.cjs'),
+        ...mustIncludeMessage,
+      },
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-unexpected-import-5.cjs'),
+        ...mustNotIncludeMessage,
+      },
+      {
+        // name: '',
+        filePath: fixtures.path('/es-modules/es-note-error-1.mjs'),
+        ...mustNotIncludeMessage,
+        errorNeedle: 'Error: some error',
```

Comment:
Let's regexes instead of strings for `errorNeedle`? That would avoid surprises if the string happens to contain some regex special char.
```suggestion
        errorNeedle: /Error: some error/,
```